### PR TITLE
Add couchdb & HTTP API for update & delete

### PR DESCRIPTION
**Pull request by aenario**
Tests & doc incoming.

### CODE REVIEW

File: couchdb/couchdb.go [hunk 1]
```diff
@@ -37,17 +37,29 @@ type JSONDoc map[string]interface{}
...
+	if ok {
+		return id
+	}
+	return ""
 }
 
 // Rev returns the revision field of the document
 func (j JSONDoc) Rev() string {
-	return j["_rev"].(string)
+	id, ok := j["_rev"].(string)
```
Comment (nono) [inline couchdb/couchdb.go]: s/id/rev/

File: couchdb/couchdb.go [hunk 2]
```diff
@@ -37,17 +37,29 @@ type JSONDoc map[string]interface{}
 
 // ID returns the identifier field of the document
```
Comment (nono) [inline couchdb/couchdb.go]: You should probably say that it is the quantified id, or give an example

File: couchdb/couchdb.go [hunk 3]
```diff
@@ -37,17 +37,29 @@ type JSONDoc map[string]interface{}
...
 func (j JSONDoc) ID() string {
-	return j["_id"].(string)
+	id, ok := j["_id"].(string)
```
Comment (nono) [inline couchdb/couchdb.go]: s/id/qid/

File: couchdb/couchdb.go [hunk 4]
```diff
@@ -154,12 +166,57 @@ func DeleteDB(dbprefix, doctype string) error {
...
-func ResetDB(dbprefix, doctype string) error {
-	err := DeleteDB(dbprefix, doctype)
-	if err != nil {
-		return err
+func ResetDB(dbprefix, doctype string) (err error) {
+	err = DeleteDB(dbprefix, doctype)
+	if err == nil {
+		return CreateDB(dbprefix, doctype)
+	}
+	return
```
Comment (nono) [inline couchdb/couchdb.go]: I'd prefer:

```go
func ResetDB(dbprefix, doctype string) error {
  err := DeleteDB(dbprefix, doctype)
  if err != nil {
    return err
  }
  return CreateDB(dbprefix, doctype)
}
```

Comment (nono) [inline couchdb/couchdb.go]: The goal is to make the code easier to change if we had to add a third step, like this:

```go
func ResetDB(dbprefix, doctype string) error {
  err := DeleteDB(dbprefix, doctype)
  if err != nil {
    return err
  }
  err = CreateDB(dbprefix, doctype)
  if err != nil {
    return err
  }
  return ThirdStep()
}
```

Comment (nono) [inline couchdb/couchdb.go]: And I think it's a golang convention to have the good case path in the final return, and errors in the if cases.

File: web/data/data_test.go [hunk 5]
```diff
@@ -136,3 +143,100 @@ func TestWrongID(t *testing.T) {
 		assert.Equal(t, "missing", out["reason"], "should give a reason")
 	}
 }
+
+// @TODO uncomment me when we stop falling back to Host = dev
```
Comment (nono) [inline web/data/data_test.go]: Maybe we can use [Skip](https://golang.org/pkg/testing/#T.Skip) for this test

File: couchdb/couchdb.go (hunk 2 above)
Comment (aenario) [inline couchdb/couchdb.go]: :+1: fixed in 9e1089c6643ac2cfee0405f0ea310abe444e1595

File: couchdb/couchdb.go (hunk 1 above)
Comment (aenario) [inline couchdb/couchdb.go]: fixed in 9e1089c6643ac2cfee0405f0ea310abe444e1595

File: couchdb/couchdb.go (hunk 4 above)
Comment (aenario) [inline couchdb/couchdb.go]: Yep, I just discovered this cool named return and may have been carried away :smile: 
Fixed in 9e1089c6643ac2cfee0405f0ea310abe444e1595

File: web/data/data_test.go (hunk 5 above)
Comment (aenario) [inline web/data/data_test.go]: Done in 9e1089c6643ac2cfee0405f0ea310abe444e1595

**Pull request merged**